cmd/zetae2e/local/utils.go

```diff
@@ -100,3 +103,52 @@ func waitKeygenHeight(
 		logger.Info("Last ZetaHeight: %d", response.Height)
 	}
 }
+
+func MonitorTxPriorityInBlocks(ctx context.Context, conf config.Config, errCh chan error) {
+	rpc, err := rpchttp.New(conf.RPCs.ZetaCoreRPC, "/websocket")
+	if err != nil {
+		errCh <- err
+	}
+
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+	for {
```

Comment:
This is a first implementation that can be refactored in the future but I still think we could already create sub function for some of the logic in the loops. The many nested loops makes it hard to read through the code